src/toxcore/dht/client.rs

```diff
@@ -125,12 +127,53 @@ impl Client {
         let nat_ping_req = DhtPacket::DhtRequest(DhtRequest::new(&self.precomputed_key.clone(), &friend_pk, &self.pk, payload));
         self.send(nat_ping_req)
     }
+    // get broadcast addresses for host's network interfaces
+    fn get_ipv4_broadcast_addrs() -> Vec<SocketAddr> {
+        let ifs = get_if_addrs::get_if_addrs().expect("no network interface");
+        ifs.iter().filter_map(|interface| 
+            match interface.addr {
+                IfAddr::V4(ref addr) => addr.broadcast,
+                _ => None,
+            })
+            .map(|ipv4|
+                SocketAddr::new(IpAddr::V4(ipv4), 33445)
+            ).collect()
+    }
+    // get broadcast addresses for Ipv4 and Ipv6
+    fn get_broadcast_addresses() -> Vec<SocketAddr> {
+        let mut res = Vec::new();
+        let ipv4s = Client::get_ipv4_broadcast_addrs();
+
+        res.extend(ipv4s.iter());
+
+        // Ipv4 global broadcast address
+        res.push(
+            SocketAddr::new(("255.255.255.255").parse().unwrap(), 33445)
```

Comment:
Why 33445? You'd better return only ip addrs without ip, set ip only when you actually send it